server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -673,7 +672,7 @@ public ClusterState execute(ClusterState currentState) {
                                 }
                             }
                         }
-                        updatedEntry = cloneEntry.withClones(clonesBuilder.build());
+                        updatedEntry = cloneEntry.withClones(clones);
```

Comment:
Wrap in Collections.unmodifiableMap.

Reply:
The `Entry` constructor calls `Map.copyOf` on this map anyway